.github/PULL_REQUEST_TEMPLATE

```diff
@@ -1,10 +1,39 @@
-## What changes were proposed in this pull request?
+<!--
+Thanks for sending a pull request!  Here are some tips for you:
+  1. If this is your first time, please read our contributor guidelines: https://spark.apache.org/contributing.html
+  2. Ensure you have added or run the appropriate tests for your PR: https://spark.apache.org/developer-tools.html
+  3. If the PR is unfinished, add '[WIP]' in your PR title, e.g., '[WIP][SPARK-XXXX] Your PR title ...'.
+  4. Update your PR description whenever it is needed.
+  5. Provide a concise example to reproduce the issue if possible for a faster review.
+-->
 
-(Please fill in changes proposed in this fix)
+### What changes were proposed in this pull request?
+<!--
+Please clarify what changes you are proposing. The purpose of this section is to outline the changes and how this PR fixes the issue. 
+Therefore, if possible, please consider write some useful notes for better and faster reviews in your PR. See the examples below.
+  1. If you refactor some codes with changing classes, showing the class hierarchy will help reviewers.
+  2. If you fix some SQL features, you can provide some references of other DBMSes.
+  3. If there is a design documentation, please leave the link.
+  4. If there is a discussion in the mailing list, please leave the link.
+-->
 
-## How was this patch tested?
 
-(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
-(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)
+### Why are the changes needed?
+<!--
+Please clarify why the changes are needed. For instance,
+  1. If you propose a new API, clarify the usecase for a new API.
+  2. If you fixes a bug, you can calrify why it is a bug.
+-->
 
-Please review https://spark.apache.org/contributing.html before opening a pull request.
+### Does this PR introduce any user-facing change?
+<!--
+If yes, please clarify the previous behavior and the change this PR proposes - provide the console output, description and/or an example to show the behaviour difference if possible.
+If no, write 'No'.
+-->
+
+
+### How was this patch tested?
+<!--
+If some tests were added, say they were added here. Please make sure to add some test cases that check the changes thoroughly including negative and positive cases if possible.
+If it was tested in a way different from regular unit tests, please clarify how you tested step by step, ideally copy and paste-able, so that other reviewers can test and check through, and descendants can verify in the future.
```

Comment:
some nits about wording:
I am not sure about the term `descendants`, but I am not a native speaker, so I hope some native English speaker can shed some light on the best term.
I'd also add that if no UT was added, it'd be great to describe why they weren't added, why it was not possible to do so.

Reply:
@srowen, does it make sense?